docs/testing-framework.md

```diff
@@ -0,0 +1,438 @@
+        return self.backend.createAccount()
+    }
+
+    /// Add a transaction to the current block.
+    ///
+    pub fun addTransaction(_ tx: Transaction) {
+        self.backend.addTransaction(tx)
+    }
+
+    /// Executes the next transaction in the block, if any.
+    /// Returns the result of the transaction, or nil if no transaction was scheduled.
+    ///
+    pub fun executeNextTransaction(): TransactionResult? {
+        return self.backend.executeNextTransaction()
+    }
+
+    /// Commit the current block.
+    /// Committing will fail if there are un-executed transactions in the block.
+    ///
+    pub fun commitBlock() {
+        self.backend.commitBlock()
+    }
+
+    /// Executes a given transaction and commit the current block.
+    ///
+    pub fun executeTransaction(_ transaction: Transaction): TransactionResult {
+        self.addTransaction(transaction)
+        let txResult = self.executeNextTransaction()!
+        self.commitBlock()
+        return txResult
+    }
+
+    /// Executes a given set of transactions and commit the current block.
+    ///
+    pub fun executeTransactions(_ transactions: [Transaction]): [TransactionResult] {
+        for tx in transactions {
+            self.addTransaction(tx)
+        }
+
+        var results: [TransactionResult] = []
+        for tx in transactions {
+            let txResult = self.executeNextTransaction()!
+            results.append(txResult)
+        }
+
+        self.commitBlock()
+        return results
+    }
+
+    /// Deploys a given contract, and initilizes it with the arguments.
+    ///
+    pub fun deployContract(
+        name: String,
+        code: String,
+        account: Account,
+        arguments: [AnyStruct]
+    ): Error? {
+        return self.backend.deployContract(
+            name: name,
+            code: code,
+            account: account,
+            arguments: arguments
+        )
+    }
+}
+```
+The `BlockchainBackend` provides the actual functionality of the blockchain.
+
+```cadence
+/// BlockchainBackend is the interface to be implemented by the backend providers.
+///
+pub struct interface BlockchainBackend {
+
+    pub fun executeScript(_ script: String, _ arguments: [AnyStruct]): ScriptResult
+
+    pub fun createAccount(): Account
+
+    pub fun addTransaction(_ transaction: Transaction)
+
+    pub fun executeNextTransaction(): TransactionResult?
+
+    pub fun commitBlock()
+
+    pub fun deployContract(
+        name: String,
+        code: String,
+        account: Account,
+        arguments: [AnyStruct]
+    ): Error?
+}
+```
+
+### Creating a blockchain
+A new blockchain instance can be created using the `newEmulatorBlockchain` method.
+It returns a `Blockchain` which is backed by a new emulator instance.
+```cadence
+let blockchain = Test.newEmulatorBlockchain()
+```
+### Creating accounts
+It may require to create accounts during tests for various reasons such as deploying contracts, signing transactions, etc.
```

Comment:
```suggestion
It may be necessary to create accounts during tests for various reasons, such as for deploying contracts, signing transactions, etc.
```